fennel/featuresets/featureset.py

```diff
@@ -311,9 +319,108 @@ def __eq__(self, other: Any) -> bool:
     def __str__(self) -> str:
         return self.name
 
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, name: str) -> None:
+        self._name = name
+        if self.extractor:
+            self.extractor.output_features = [name]
+
+    @property
+    def featureset_name(self) -> str:
+        return self._featureset_name
+
+    @featureset_name.setter
+    def featureset_name(self, name: str) -> None:
+        self._featureset_name = name
+        if self.extractor:
+            self.extractor.featureset = name
+
     def fqn(self) -> str:
         return self.fqn_
 
+    def extract(
+        self,
+        field: Field = None,
+        provider: Featureset = None,
+        default=None,
+        feature: Feature = None,
+        version: int = 0,
+    ) -> Feature:
+        """
+        Derives an extractor for the feature using the given params.
+        The derived extractor either performs a lookup on the given field,
+        or is an alias to the given feature.
+
+        Parameters:
+        field: the field in a dataset to lookup and return
+        provider: the input featureset that should contain a feature matching
+                  the field name. If not provided, then the current featureset
+                  is assumed to contain the respective feature.
+        default: An optional default value to fill null values from the lookup
+        feature: If provided, this function creates a one way alias from the
+                    calling feature to this feature.
+        version: the version of this extractor
+
+        Returns:
+        Feature: This feature
+        """
+        if self.extractor:
+            raise TypeError(
+                f"extract() can only be called once for feature id={self.id}"
+            )
+        if field is None and feature is None:
+            raise TypeError(
+                f"Either field or feature must be specified to extract feature id={self.id}"
+            )
+
+        # aliasing
+        if feature:
+            self.extractor = Extractor(
+                name=f"alias_{feature}",
+                extractor_type=ExtractorType.ALIAS,
+                inputs=[feature],
+                outputs=[self.id],
+                version=version,
+            )
+            return self
+
+        provider_features = []
+        # If provider is none, then the provider is this featureset. The input features
+        # are captured once this featureset is initialized
+        name = f"lookup_{field}"
+        field = cast(Field, field)
+        ds = None
+        if provider:
+            if hasattr(field, "dataset"):
+                ds = field.dataset
+            if not ds:
+                raise ValueError(
+                    f"Dataset {field.dataset_name} not found for field {field}"
+                )
```

Comment:
if not hasttra(field, "dataset")
   raise ..
   
ds = getattr(... ), 

you dont need to define ds=None and get rid of an if

Reply:
Changed to `if/else`. I still am assigning `ds` to a var so avoid having to call `hasattr` again on Line 420